src/crsAdapter/MerlinAdapter.js

```diff
@@ -255,26 +262,43 @@ class MerlinAdapter {
      * @param xml object
      */
     assignCarServiceFromAdapterObjectToXmlObject(service, xmlService, xml) {
-        const calculateDropOffDate = (service) => {
-            if (service.dropOffDate) {
-                let dropOffDate = moment(service.dropOffDate, this.options.useDateFormat);
-
-                return dropOffDate.isValid() ? dropOffDate.format(CONFIG.crs.dateFormat) : service.dropOffDate;
-            }
-
-            let pickUpDate = moment(service.pickUpDate, this.options.useDateFormat);
-
-            return pickUpDate.isValid()
-                ? pickUpDate.add(service.duration, 'days').format(CONFIG.crs.dateFormat)
-                : service.pickUpDate;
-        };
-
         const reduceExtrasList = (extras) => {
             return (extras || []).join('|')
                 .replace(/childCareSeat0/g, 'BS')
                 .replace(/childCareSeat((\d){1,2})/g, 'CS$1YRS');
         };
 
+        let pickUpDate = moment(service.pickUpDate, this.options.useDateFormat);
```

Comment:
The date calculations look very pretty now :) Why id you get rid of the local function though?

Reply:
because for the calculation of the dropoff date there is no function needed anymore. i will check if this logic is used more than once and create a function for it when it is the case.